core/src/test/scala/unit/kafka/log/LogTest.scala

```diff
@@ -199,6 +199,20 @@ class LogTest {
   }
 
   @Test
+  def testSizeForLargeLogs(): Unit = {
+    val largeSize = Int.MaxValue.toLong * 2
+    val logSegment = EasyMock.createMock(classOf[LogSegment])
+
+    EasyMock.expect(logSegment.size).andReturn(Int.MaxValue)
+    EasyMock.expectLastCall().anyTimes()
```

Comment:
Nit: you can add `anyTimes` on the line above. I can do that while merging as well.